cmd/tailscale/cli/serve_dev.go

```diff
@@ -83,36 +132,86 @@ func newServeDevCommand(e *serveEnv, subcmd string) *ffcli.Command {
 	}
 }
 
-// runServeDev is the entry point for the "tailscale {serve,funnel}" commands.
-func (e *serveEnv) runServeDev(funnel bool) execFunc {
+// runServeCombined is the entry point for the "tailscale {serve,funnel}" commands.
+func (e *serveEnv) runServeCombined(subcmd serveMode) execFunc {
+	e.subcmd = subcmd
+
 	return func(ctx context.Context, args []string) error {
-		ctx, cancel := signal.NotifyContext(ctx, os.Interrupt)
-		defer cancel()
-		if len(args) != 1 {
-			return flag.ErrHelp
+		funnel := subcmd == funnel
+
+		err := checkLegacyInvocation(subcmd, args)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: the CLI for serve and funnel has changed.\n")
+			fmt.Fprintf(os.Stderr, "Please see https://tailscale.com/kb/1242/tailscale-serve for more information.\n\n")
+
+			return errHelp
 		}
-		var source string
-		port64, err := strconv.ParseUint(args[0], 10, 16)
-		if err == nil {
-			source = fmt.Sprintf("http://127.0.0.1:%d", port64)
-		} else {
-			source, err = expandProxyTarget(args[0])
+
+		if len(args) > 2 {
+			fmt.Fprintf(os.Stderr, "error: invalid number of arguments (%d)\n\n", len(args))
+			return errHelp
 		}
-		if err != nil {
-			return err
+
+		// always run in the background when using --set-path
+		if e.setPath != "" {
+			e.bg = true
 		}
 
+		turnOff := "off" == args[len(args)-1]
```

Comment:
I ran `tailscale funnel --bg 4545` and worked nicely
But then I just put an "off" at the end and got the "legacy" message.